modepy/quadrature/__init__.py

```diff
@@ -46,48 +51,55 @@ class QuadratureRuleUnavailable(RuntimeError):
 
 
 class Quadrature:
-    """The basic interface for a quadrature rule.
+    """The basic interface for a quadrature rule."""
 
-    .. attribute :: nodes
+    def __init__(self,
+                 nodes: np.ndarray,
+                 weights: np.ndarray,
+                 exact_to: Optional[int] = None) -> None:
+        """
+        :arg nodes: an array of shape *(d, nnodes)*, where *d* is the dimension
+            of the qudrature rule.
+        :arg weights: an array of length *nnodes*.
+        :arg exact_to: an optional argument denoting the summed polynomial
+            degree to which the quadrature is exact. By default, `exact_to`
+            is `None` and will *not* be set as an attribute.
+        """
 
-        An array of shape *(d, nnodes)*, where *d* is the dimension
+        self.nodes: np.ndarray = nodes
+        """An array of shape *(dim, nnodes)*, where *dim* is the dimension
         of the qudrature rule. In 1D, the shape is just *(nnodes,)*.
+        """
```

Comment:
:exploding_head: Sphinx reads the constructor source?

Reply:
Yes it does! :grin: I'm not sure when all this magic got in there, but it seems to work.